In [12]:
import pandas as pd
import numpy as np

In [5]:
df = pd.read_csv('1698635826265-telco_customer_churn.csv')
pd.set_option("display.max_columns",None)

In [6]:
df

customerID  gender  SeniorCitizen Partner Dependents  tenure  \
0     7590-VHVEG  Female              0     Yes         No       1   
1     5575-GNVDE    Male              0      No         No      34   
2     3668-QPYBK    Male              0      No         No       2   
3     7795-CFOCW    Male              0      No         No      45   
4     9237-HQITU  Female              0      No         No       2   
...          ...     ...            ...     ...        ...     ...   
7038  6840-RESVB    Male              0     Yes        Yes      24   
7039  2234-XADUH  Female              0     Yes        Yes      72   
7040  4801-JZAZL  Female              0     Yes        Yes      11   
7041  8361-LTMKD    Male              1     Yes         No       4   
7042  3186-AJIEK    Male              0      No         No      66   

     PhoneService     MultipleLines InternetService OnlineSecurity  \
0              No  No phone service             DSL             No   
1             Yes                No             DSL            Yes   
2             Yes                No             DSL            Yes   
3              No  No phone service             DSL            Yes   
4             Yes                No     Fiber optic             No   
...           ...               ...             ...            ...   
7038          Yes               Yes             DSL            Yes   
7039          Yes               Yes     Fiber optic             No   
7040           No  No phone service             DSL            Yes   
7041          Yes               Yes     Fiber optic             No   
7042          Yes                No     Fiber optic            Yes   

     OnlineBackup DeviceProtection TechSupport StreamingTV StreamingMovies  \
0             Yes               No          No          No              No   
1              No              Yes          No          No              No   
2             Yes               No          No          No              No   
3              No              Yes         Yes          No              No   
4              No               No          No          No              No   
...           ...              ...         ...         ...             ...   
7038           No              Yes         Yes         Yes             Yes   
7039          Yes              Yes          No         Yes             Yes   
7040           No               No          No          No              No   
7041           No               No          No          No              No   
7042           No              Yes         Yes         Yes             Yes   

            Contract PaperlessBilling              PaymentMethod  \
0     Month-to-month              Yes           Electronic check   
1           One year               No               Mailed check   
2     Month-to-month              Yes               Mailed check   
3           One year               No  Bank transfer (automatic)   
4     Month-to-month              Yes           Electronic check   
...              ...              ...                        ...   
7038        One year              Yes               Mailed check   
7039        One year              Yes    Credit card (automatic)   
7040  Month-to-month              Yes           Electronic check   
7041  Month-to-month              Yes               Mailed check   
7042        Two year              Yes  Bank transfer (automatic)   

      MonthlyCharges TotalCharges Churn  
0              29.85        29.85    No  
1              56.95       1889.5    No  
2              53.85       108.15   Yes  
3              42.30      1840.75    No  
4              70.70       151.65   Yes  
...              ...          ...   ...  
7038           84.80       1990.5    No  
7039          103.20       7362.9    No  
7040           29.60       346.45    No  
7041           74.40        306.6   Yes  
7042          105.65       6844.5    No  

[7043 rows x 21 columns]

In [8]:
data = df.copy()
data

customerID  gender  SeniorCitizen Partner Dependents  tenure  \
0     7590-VHVEG  Female              0     Yes         No       1   
1     5575-GNVDE    Male              0      No         No      34   
2     3668-QPYBK    Male              0      No         No       2   
3     7795-CFOCW    Male              0      No         No      45   
4     9237-HQITU  Female              0      No         No       2   
...          ...     ...            ...     ...        ...     ...   
7038  6840-RESVB    Male              0     Yes        Yes      24   
7039  2234-XADUH  Female              0     Yes        Yes      72   
7040  4801-JZAZL  Female              0     Yes        Yes      11   
7041  8361-LTMKD    Male              1     Yes         No       4   
7042  3186-AJIEK    Male              0      No         No      66   

     PhoneService     MultipleLines InternetService OnlineSecurity  \
0              No  No phone service             DSL             No   
1             Yes                No             DSL            Yes   
2             Yes                No             DSL            Yes   
3              No  No phone service             DSL            Yes   
4             Yes                No     Fiber optic             No   
...           ...               ...             ...            ...   
7038          Yes               Yes             DSL            Yes   
7039          Yes               Yes     Fiber optic             No   
7040           No  No phone service             DSL            Yes   
7041          Yes               Yes     Fiber optic             No   
7042          Yes                No     Fiber optic            Yes   

     OnlineBackup DeviceProtection TechSupport StreamingTV StreamingMovies  \
0             Yes               No          No          No              No   
1              No              Yes          No          No              No   
2             Yes               No          No          No              No   
3              No              Yes         Yes          No              No   
4              No               No          No          No              No   
...           ...              ...         ...         ...             ...   
7038           No              Yes         Yes         Yes             Yes   
7039          Yes              Yes          No         Yes             Yes   
7040           No               No          No          No              No   
7041           No               No          No          No              No   
7042           No              Yes         Yes         Yes             Yes   

            Contract PaperlessBilling              PaymentMethod  \
0     Month-to-month              Yes           Electronic check   
1           One year               No               Mailed check   
2     Month-to-month              Yes               Mailed check   
3           One year               No  Bank transfer (automatic)   
4     Month-to-month              Yes           Electronic check   
...              ...              ...                        ...   
7038        One year              Yes               Mailed check   
7039        One year              Yes    Credit card (automatic)   
7040  Month-to-month              Yes           Electronic check   
7041  Month-to-month              Yes               Mailed check   
7042        Two year              Yes  Bank transfer (automatic)   

      MonthlyCharges TotalCharges Churn  
0              29.85        29.85    No  
1              56.95       1889.5    No  
2              53.85       108.15   Yes  
3              42.30      1840.75    No  
4              70.70       151.65   Yes  
...              ...          ...   ...  
7038           84.80       1990.5    No  
7039          103.20       7362.9    No  
7040           29.60       346.45    No  
7041           74.40        306.6   Yes  
7042          105.65       6844.5    No  

[7043 rows x 21 columns]

### Welcome to Homework. Please answer the questions below!

Score : Maximal 50.
Per Soal bobotnya 10

1. Buanglah baris dengan `tenure` samadengan 0 (jadi, data yang akan kita analisa di PR ini adalah baris-baris dengan `tenure` TIDAK samadengan 0). Ada berapa baris yang sekarang tersedia di data kita?

2. Bagi customer yang ***TIDAK MEMILIKI PARTNER*** dan ***TIDAK MEMILIKI DEPENDENTS***, berapa ***rata-rata*** dari `tenure` mereka?

3. Bagi customer yang ***MEMILIKI PARTNER*** dan ***TIDAK MEMILIKI DEPENDENTS***, berapa maksimal `MonthlyCharges` yang tercata?

4. Berapa banyak customer yang ***TIDAK MEMILIKI PARTNER*** namun ***MEMILIKI DEPENDENTS***?

5. Berapa rata-rata ***TOTAL CHARGES*** dari customer ***NON-SENIOR CITIZEN*** (Senior Citizen = 0) yang memiliki ***PHONE SERVICE*** dan memiliki ***INTERNET SERVICE FIBER OPTIC***?

#### Silakan menuliskan kode untuk menjawab 5 pertanyaan di atas...

In [9]:
data.shape

(7043, 21)

In [10]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7043 entries, 0 to 7042
Data columns (total 21 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   customerID        7043 non-null   object 
 1   gender            7043 non-null   object 
 2   SeniorCitizen     7043 non-null   int64  
 3   Partner           7043 non-null   object 
 4   Dependents        7043 non-null   object 
 5   tenure            7043 non-null   int64  
 6   PhoneService      7043 non-null   object 
 7   MultipleLines     7043 non-null   object 
 8   InternetService   7043 non-null   object 
 9   OnlineSecurity    7043 non-null   object 
 10  OnlineBackup      7043 non-null   object 
 11  DeviceProtection  7043 non-null   object 
 12  TechSupport       7043 non-null   object 
 13  StreamingTV       7043 non-null   object 
 14  StreamingMovies   7043 non-null   object 
 15  Contract          7043 non-null   object 
 16  PaperlessBilling  7043 non-null   object 


In [50]:
data_hapus = data.loc[data['tenure'] != 0 ]
print(data_hapus)

      customerID  gender  SeniorCitizen Partner Dependents  tenure  \
0     7590-VHVEG  Female              0     Yes         No       1   
1     5575-GNVDE    Male              0      No         No      34   
2     3668-QPYBK    Male              0      No         No       2   
3     7795-CFOCW    Male              0      No         No      45   
4     9237-HQITU  Female              0      No         No       2   
...          ...     ...            ...     ...        ...     ...   
7038  6840-RESVB    Male              0     Yes        Yes      24   
7039  2234-XADUH  Female              0     Yes        Yes      72   
7040  4801-JZAZL  Female              0     Yes        Yes      11   
7041  8361-LTMKD    Male              1     Yes         No       4   
7042  3186-AJIEK    Male              0      No         No      66   

     PhoneService     MultipleLines InternetService OnlineSecurity  \
0              No  No phone service             DSL             No   
1             Yes  

In [64]:
data_hapus.shape

(7032, 21)

In [75]:
kondisi = data[(data['Partner'] == 'No') & (data['Dependents'] == 'No')]

kondisi['tenure'].mean()

23.303353658536587

In [76]:
kondisi2 = data[(data['Partner'] == 'Yes') & (data['Dependents'] == 'No')]

kondisi2['MonthlyCharges'].max()

118.6

In [77]:
kondisi2.sort_values(by='MonthlyCharges', ascending=False)

customerID  gender  SeniorCitizen Partner Dependents  tenure  \
3894  5989-AXPUC  Female              0     Yes         No      68   
4610  2889-FPWRM    Male              0     Yes         No      72   
4875  2302-ANTDP  Female              1     Yes         No      48   
2368  6650-BWFRT  Female              0     Yes         No      72   
5347  9788-HNGUT    Male              0     Yes         No      72   
...          ...     ...            ...     ...        ...     ...   
4927  4774-HHGGS    Male              0     Yes         No      30   
4304  9732-OUYRN  Female              0     Yes         No      49   
6178  3719-TDVQB  Female              1     Yes         No      54   
6925  6508-NJYRO    Male              0     Yes         No      15   
1529  9764-REAFF  Female              0     Yes         No      59   

     PhoneService MultipleLines InternetService       OnlineSecurity  \
3894          Yes           Yes     Fiber optic                  Yes   
4610          Yes           Yes     Fiber optic                  Yes   
4875          Yes           Yes     Fiber optic                  Yes   
2368          Yes           Yes     Fiber optic                  Yes   
5347          Yes           Yes     Fiber optic                  Yes   
...           ...           ...             ...                  ...   
4927          Yes            No              No  No internet service   
4304          Yes            No              No  No internet service   
6178          Yes            No              No  No internet service   
6925          Yes            No              No  No internet service   
1529          Yes            No              No  No internet service   

             OnlineBackup     DeviceProtection          TechSupport  \
3894                  Yes                  Yes                  Yes   
4610                  Yes                  Yes                  Yes   
4875                  Yes                  Yes                  Yes   
2368                  Yes                  Yes                  Yes   
5347                  Yes                  Yes                  Yes   
...                   ...                  ...                  ...   
4927  No internet service  No internet service  No internet service   
4304  No internet service  No internet service  No internet service   
6178  No internet service  No internet service  No internet service   
6925  No internet service  No internet service  No internet service   
1529  No internet service  No internet service  No internet service   

              StreamingTV      StreamingMovies        Contract  \
3894                  Yes                  Yes        Two year   
4610                  Yes                  Yes        One year   
4875                  Yes                  Yes  Month-to-month   
2368                  Yes                  Yes        Two year   
5347                  Yes                  Yes        Two year   
...                   ...                  ...             ...   
4927  No internet service  No internet service        Two year   
4304  No internet service  No internet service        One year   
6178  No internet service  No internet service        Two year   
6925  No internet service  No internet service        One year   
1529  No internet service  No internet service        Two year   

     PaperlessBilling              PaymentMethod  MonthlyCharges TotalCharges  \
3894               No               Mailed check          118.60      7990.05   
4610              Yes  Bank transfer (automatic)          117.80       8684.8   
4875               No           Electronic check          117.45       5438.9   
2368              Yes    Credit card (automatic)          117.15       8529.5   
5347              Yes    Credit card (automatic)          116.95       8594.4   
...               ...                        ...             ...          ...   
4927               No               Mailed check           19.05       637.55   
4304  

In [84]:
kondisi3 = data[(data['Partner'] == 'No') & (data['Dependents'] == 'Yes')]
kondisi3.shape

(361, 21)

In [87]:
kondisi3['customerID'].nunique()

361

In [91]:
data = data[data["TotalCharges"]!=" "]

In [94]:
data.info()

<class 'pandas.core.frame.DataFrame'>
Index: 7032 entries, 0 to 7042
Data columns (total 21 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   customerID        7032 non-null   object 
 1   gender            7032 non-null   object 
 2   SeniorCitizen     7032 non-null   int64  
 3   Partner           7032 non-null   object 
 4   Dependents        7032 non-null   object 
 5   tenure            7032 non-null   int64  
 6   PhoneService      7032 non-null   object 
 7   MultipleLines     7032 non-null   object 
 8   InternetService   7032 non-null   object 
 9   OnlineSecurity    7032 non-null   object 
 10  OnlineBackup      7032 non-null   object 
 11  DeviceProtection  7032 non-null   object 
 12  TechSupport       7032 non-null   object 
 13  StreamingTV       7032 non-null   object 
 14  StreamingMovies   7032 non-null   object 
 15  Contract          7032 non-null   object 
 16  PaperlessBilling  7032 non-null   object 
 17  

In [106]:
data['TotalCharges'] = data['TotalCharges'].astype(float)

In [115]:
data.info()

<class 'pandas.core.frame.DataFrame'>
Index: 7032 entries, 0 to 7042
Data columns (total 21 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   customerID        7032 non-null   object 
 1   gender            7032 non-null   object 
 2   SeniorCitizen     7032 non-null   int64  
 3   Partner           7032 non-null   object 
 4   Dependents        7032 non-null   object 
 5   tenure            7032 non-null   int64  
 6   PhoneService      7032 non-null   object 
 7   MultipleLines     7032 non-null   object 
 8   InternetService   7032 non-null   object 
 9   OnlineSecurity    7032 non-null   object 
 10  OnlineBackup      7032 non-null   object 
 11  DeviceProtection  7032 non-null   object 
 12  TechSupport       7032 non-null   object 
 13  StreamingTV       7032 non-null   object 
 14  StreamingMovies   7032 non-null   object 
 15  Contract          7032 non-null   object 
 16  PaperlessBilling  7032 non-null   object 
 17  

In [116]:
data

customerID  gender  SeniorCitizen Partner Dependents  tenure  \
0     7590-VHVEG  Female              0     Yes         No       1   
1     5575-GNVDE    Male              0      No         No      34   
2     3668-QPYBK    Male              0      No         No       2   
3     7795-CFOCW    Male              0      No         No      45   
4     9237-HQITU  Female              0      No         No       2   
...          ...     ...            ...     ...        ...     ...   
7038  6840-RESVB    Male              0     Yes        Yes      24   
7039  2234-XADUH  Female              0     Yes        Yes      72   
7040  4801-JZAZL  Female              0     Yes        Yes      11   
7041  8361-LTMKD    Male              1     Yes         No       4   
7042  3186-AJIEK    Male              0      No         No      66   

     PhoneService     MultipleLines InternetService OnlineSecurity  \
0              No  No phone service             DSL             No   
1             Yes                No             DSL            Yes   
2             Yes                No             DSL            Yes   
3              No  No phone service             DSL            Yes   
4             Yes                No     Fiber optic             No   
...           ...               ...             ...            ...   
7038          Yes               Yes             DSL            Yes   
7039          Yes               Yes     Fiber optic             No   
7040           No  No phone service             DSL            Yes   
7041          Yes               Yes     Fiber optic             No   
7042          Yes                No     Fiber optic            Yes   

     OnlineBackup DeviceProtection TechSupport StreamingTV StreamingMovies  \
0             Yes               No          No          No              No   
1              No              Yes          No          No              No   
2             Yes               No          No          No              No   
3              No              Yes         Yes          No              No   
4              No               No          No          No              No   
...           ...              ...         ...         ...             ...   
7038           No              Yes         Yes         Yes             Yes   
7039          Yes              Yes          No         Yes             Yes   
7040           No               No          No          No              No   
7041           No               No          No          No              No   
7042           No              Yes         Yes         Yes             Yes   

            Contract PaperlessBilling              PaymentMethod  \
0     Month-to-month              Yes           Electronic check   
1           One year               No               Mailed check   
2     Month-to-month              Yes               Mailed check   
3           One year               No  Bank transfer (automatic)   
4     Month-to-month              Yes           Electronic check   
...              ...              ...                        ...   
7038        One year              Yes               Mailed check   
7039        One year              Yes    Credit card (automatic)   
7040  Month-to-month              Yes           Electronic check   
7041  Month-to-month              Yes               Mailed check   
7042        Two year              Yes  Bank transfer (automatic)   

      MonthlyCharges  TotalCharges Churn  
0              29.85         29.85    No  
1              56.95       1889.50    No  
2              53.85        108.15   Yes  
3              42.30       1840.75    No  
4              70.70        151.65   Yes  
...              ...           ...   ...  
7038           84.80       1990.50    No  
7039          103.20       7362.90    No  
7040           29.60        346.45    No  
7041           74.40        306.60   Yes  
7042          105.65       6844.50    No  

[7032 rows x 21 columns]

In [119]:
kondisi4 = data[(data['SeniorCitizen'] == 0) 
                & (data['PhoneService'] == 'Yes') 
                & (data['InternetService'] == 'Fiber optic')]

kondisi4['TotalCharges'].mean()

3226.4927373068435

### Silakan menjawab di bawah ini:

#### Soal 1:
Buanglah baris dengan tenure samadengan 0 (jadi, data yang akan kita analisa di PR ini adalah baris-baris dengan tenure TIDAK samadengan 0). Ada berapa baris yang sekarang tersedia di data kita?

#### Jawaban: 7032 (langsung edit di sini aja)

#### Soal 2:
Bagi customer yang TIDAK MEMILIKI PARTNER dan TIDAK MEMILIKI DEPENDENTS, berapa rata-rata dari tenure mereka?

#### Jawaban: 23.30

#### Soal 3:
Bagi customer yang MEMILIKI PARTNER dan TIDAK MEMILIKI DEPENDENTS, berapa maksimal MonthlyCharges yang tercata?

#### Jawaban: 118.6

#### Soal 4:
Berapa banyak customer yang TIDAK MEMILIKI PARTNER namun MEMILIKI DEPENDENTS?

#### Jawaban: 361

#### Soal 5:
Berapa rata-rata TOTAL CHARGES dari customer NON-SENIOR CITIZEN (Senior Citizen = 0) yang memiliki PHONE SERVICE dan memiliki INTERNET SERVICE FIBER OPTIC?

#### Jawaban: 3226.49

# Deep Data Manipulation

Lakukan analisa dengan menggunakan teknik teknik data manipulation yang telah dipelajari sebanyak 2 analisa. kemudian berikan intrepetasi dan alasan anda melakukan analisa tersebut. <br>

Misalnya : <br>
Melakukan filter monthly charges diatas 500 kemudian saya akan mengalisa percentase churn berdasrkan kelompok A. Hasil temuan, ternyata kelompok tersebut memiliki percent churn sebesar %. <br>

Bobot maksimal 50 <br>
semakian baik analisa dan intrepetasi disertai alasan yang jelas akan mendapatkan score maximal

# Bagaimana pengaruh jumlah total charges pada 
* kelompok A(tidak memiliki tanggungan dan tidak memiliki partner), 
* kelompok B (memiliki tanggungan dan tidak memiliki partner),
* kelompok C (memiliki tanggungan dan memiliki partner) dan
* kelompok D(tidak memiliki tanggungan dan memiliki partner)?

In [162]:
kel_a = data[(data['Dependents'] == 'No') & (data['Partner'] == 'No')]
kel_b = data[(data['Dependents'] == 'Yes') & (data['Partner'] == 'No')]
kel_c = data[(data['Dependents'] == 'Yes') & (data['Partner'] == 'Yes')]
kel_d = data[(data['Dependents'] == 'No') & (data['Partner'] == 'Yes')]

In [147]:
kel_a

customerID  gender  SeniorCitizen Partner Dependents  tenure  \
1     5575-GNVDE    Male              0      No         No      34   
2     3668-QPYBK    Male              0      No         No       2   
3     7795-CFOCW    Male              0      No         No      45   
4     9237-HQITU  Female              0      No         No       2   
5     9305-CDSKC  Female              0      No         No       8   
...          ...     ...            ...     ...        ...     ...   
7034  0639-TSIQW  Female              0      No         No      67   
7035  8456-QDAVC    Male              0      No         No      19   
7036  7750-EYXWZ  Female              0      No         No      12   
7037  2569-WGERO  Female              0      No         No      72   
7042  3186-AJIEK    Male              0      No         No      66   

     PhoneService     MultipleLines InternetService       OnlineSecurity  \
1             Yes                No             DSL                  Yes   
2             Yes                No             DSL                  Yes   
3              No  No phone service             DSL                  Yes   
4             Yes                No     Fiber optic                   No   
5             Yes               Yes     Fiber optic                   No   
...           ...               ...             ...                  ...   
7034          Yes               Yes     Fiber optic                  Yes   
7035          Yes                No     Fiber optic                   No   
7036           No  No phone service             DSL                   No   
7037          Yes                No              No  No internet service   
7042          Yes                No     Fiber optic                  Yes   

             OnlineBackup     DeviceProtection          TechSupport  \
1                      No                  Yes                   No   
2                     Yes                   No                   No   
3                      No                  Yes                  Yes   
4                      No                   No                   No   
5                      No                  Yes                   No   
...                   ...                  ...                  ...   
7034                  Yes                  Yes                   No   
7035                   No                   No                   No   
7036                  Yes                  Yes                  Yes   
7037  No internet service  No internet service  No internet service   
7042                   No                  Yes                  Yes   

              StreamingTV      StreamingMovies        Contract  \
1                      No                   No        One year   
2                      No                   No  Month-to-month   
3                      No                   No        One year   
4                      No                   No  Month-to-month   
5                     Yes                  Yes  Month-to-month   
...                   ...                  ...             ...   
7034                  Yes                   No  Month-to-month   
7035                  Yes                   No  Month-to-month   
7036                  Yes                  Yes        One year   
7037  No internet service  No internet service        Two year   
7042                  Yes                  Yes        Two year   

     PaperlessBilling              PaymentMethod  MonthlyCharges  \
1                  No               Mailed check           56.95   
2                 Yes               Mailed check           53.85   
3                  No  Bank transfer (automatic)           42.30   
4                 Yes           Electronic check           70.70   
5                 Yes           Electronic check           99.65   
...               ...                        ...             ...   
7034              Yes    Credit card (automatic)          102.95   
7035              Yes  Bank transfer (automatic)           78.

In [156]:
kel_a['TotalCharges'].sum()

5283104.0

In [161]:
kel_a['customerID'].nunique()

3280

In [149]:
kel_b

customerID  gender  SeniorCitizen Partner Dependents  tenure  \
6     1452-KIOVK    Male              0      No        Yes      22   
9     6388-TABGU    Male              0      No        Yes      62   
17    9959-WOFKT    Male              0      No        Yes      71   
29    8773-HHUOZ  Female              0      No        Yes      17   
74    5630-AHZIL  Female              0      No        Yes       3   
...          ...     ...            ...     ...        ...     ...   
6946  9013-AQORL  Female              0      No        Yes      48   
6966  6598-KELSS    Male              0      No        Yes      50   
6979  5351-QESIO    Male              0      No        Yes       1   
6984  6447-GORXK    Male              0      No        Yes      31   
6995  7164-BPTUT    Male              0      No        Yes      57   

     PhoneService     MultipleLines InternetService OnlineSecurity  \
6             Yes               Yes     Fiber optic             No   
9             Yes                No             DSL            Yes   
17            Yes               Yes     Fiber optic            Yes   
29            Yes                No             DSL             No   
74            Yes                No             DSL            Yes   
...           ...               ...             ...            ...   
6946           No  No phone service             DSL             No   
6966           No  No phone service             DSL            Yes   
6979           No  No phone service             DSL             No   
6984           No  No phone service             DSL             No   
6995          Yes               Yes             DSL            Yes   

     OnlineBackup DeviceProtection TechSupport StreamingTV StreamingMovies  \
6             Yes               No          No         Yes              No   
9             Yes               No          No          No              No   
17             No              Yes          No         Yes             Yes   
29             No               No          No         Yes             Yes   
74             No               No         Yes          No             Yes   
...           ...              ...         ...         ...             ...   
6946          Yes              Yes          No          No             Yes   
6966          Yes               No          No         Yes              No   
6979           No               No          No          No              No   
6984           No               No          No          No              No   
6995          Yes              Yes         Yes         Yes             Yes   

            Contract PaperlessBilling              PaymentMethod  \
6     Month-to-month              Yes    Credit card (automatic)   
9           One year               No  Bank transfer (automatic)   
17          Two year               No  Bank transfer (automatic)   
29    Month-to-month              Yes               Mailed check   
74    Month-to-month              Yes  Bank transfer (automatic)   
...              ...              ...                        ...   
6946  Month-to-month               No    Credit card (automatic)   
6966        One year              Yes  Bank transfer (automatic)   
6979  Month-to-month               No               Mailed check   
6984  Month-to-month               No    Credit card (automatic)   
6995        Two year               No               Mailed check   

      MonthlyCharges  TotalCharges Churn  
6              89.10       1949.40    No  
9              56.15       3487.95    No  
17            106.70       7382.25    No  
29             64.70       1093.10   Yes  
74             64.50        177.40    No  
...              ...           ...   ...  
6946           45.55       2108.35    No  
6966           43.05       2208.05    No  
6979           24.20         24.20    No  
6984           25.00        789.20    No  
6995           89.55       5012.35    No  

[359 rows x 21 columns]

In [157]:
kel_b['TotalCharges'].sum()

484567.0

In [160]:
kel_b['customerID'].nunique()

359

In [154]:
kel_c

customerID  gender  SeniorCitizen Partner Dependents  tenure  \
10    9763-GRSKD    Male              0     Yes        Yes      13   
15    3655-SNQYZ  Female              0     Yes        Yes      69   
18    4190-MFLUW  Female              0     Yes        Yes      10   
24    6322-HRPFA    Male              0     Yes        Yes      49   
26    6467-CHFZW    Male              0     Yes        Yes      47   
...          ...     ...            ...     ...        ...     ...   
7018  1122-JWTJW    Male              0     Yes        Yes       1   
7020  9837-FWLCH    Male              0     Yes        Yes      12   
7038  6840-RESVB    Male              0     Yes        Yes      24   
7039  2234-XADUH  Female              0     Yes        Yes      72   
7040  4801-JZAZL  Female              0     Yes        Yes      11   

     PhoneService     MultipleLines InternetService       OnlineSecurity  \
10            Yes                No             DSL                  Yes   
15            Yes               Yes     Fiber optic                  Yes   
18            Yes                No             DSL                   No   
24            Yes                No             DSL                  Yes   
26            Yes               Yes     Fiber optic                   No   
...           ...               ...             ...                  ...   
7018          Yes                No     Fiber optic                   No   
7020          Yes                No              No  No internet service   
7038          Yes               Yes             DSL                  Yes   
7039          Yes               Yes     Fiber optic                   No   
7040           No  No phone service             DSL                  Yes   

             OnlineBackup     DeviceProtection          TechSupport  \
10                     No                   No                   No   
15                    Yes                  Yes                  Yes   
18                     No                  Yes                  Yes   
24                    Yes                   No                  Yes   
26                    Yes                   No                   No   
...                   ...                  ...                  ...   
7018                   No                   No                   No   
7020  No internet service  No internet service  No internet service   
7038                   No                  Yes                  Yes   
7039                  Yes                  Yes                   No   
7040                   No                   No                   No   

              StreamingTV      StreamingMovies        Contract  \
10                     No                   No  Month-to-month   
15                    Yes                  Yes        Two year   
18                     No                   No  Month-to-month   
24                     No                   No  Month-to-month   
26                    Yes                  Yes  Month-to-month   
...                   ...                  ...             ...   
7018                   No                   No  Month-to-month   
7020  No internet service  No internet service  Month-to-month   
7038                  Yes                  Yes        One year   
7039                  Yes                  Yes        One year   
7040                   No                   No  Month-to-month   

     PaperlessBilling            PaymentMethod  MonthlyCharges  TotalCharges  \
10                Yes             Mailed check           49.95        587.45   
15                 No  Credit card (automatic)          113.25       7895.15   
18                 No  Credit card (automatic)           55.20        528.35   
24                 No  Credit card (automatic)           59.60       2970.30   
26                Yes         Electronic check           99.35       4749.15   
...               ...                      ...             ...           ...   
7018              Yes             Mailed check

In [158]:
kel_c['TotalCharges'].sum()

4779631.95

In [159]:
kel_c['customerID'].nunique()

1740

In [163]:
kel_d

customerID  gender  SeniorCitizen Partner Dependents  tenure  \
0     7590-VHVEG  Female              0     Yes         No       1   
8     7892-POOKP  Female              0     Yes         No      28   
12    8091-TTVAX    Male              0     Yes         No      58   
21    1680-VDCWW    Male              0     Yes         No      12   
23    3638-WEABW  Female              0     Yes         No      58   
...          ...     ...            ...     ...        ...     ...   
7023  1035-IPQPU  Female              1     Yes         No      63   
7024  7398-LXGYX    Male              0     Yes         No      44   
7028  9281-CEDRU  Female              0     Yes         No      68   
7031  3605-JISKB    Male              1     Yes         No      55   
7041  8361-LTMKD    Male              1     Yes         No       4   

     PhoneService     MultipleLines InternetService       OnlineSecurity  \
0              No  No phone service             DSL                   No   
8             Yes               Yes     Fiber optic                   No   
12            Yes               Yes     Fiber optic                   No   
21            Yes                No              No  No internet service   
23            Yes               Yes             DSL                   No   
...           ...               ...             ...                  ...   
7023          Yes               Yes     Fiber optic                   No   
7024          Yes               Yes     Fiber optic                  Yes   
7028          Yes                No             DSL                   No   
7031          Yes               Yes             DSL                  Yes   
7041          Yes               Yes     Fiber optic                   No   

             OnlineBackup     DeviceProtection          TechSupport  \
0                     Yes                   No                   No   
8                      No                  Yes                  Yes   
12                     No                  Yes                   No   
21    No internet service  No internet service  No internet service   
23                    Yes                   No                  Yes   
...                   ...                  ...                  ...   
7023                  Yes                  Yes                   No   
7024                   No                  Yes                   No   
7028                  Yes                   No                  Yes   
7031                  Yes                   No                   No   
7041                   No                   No                   No   

              StreamingTV      StreamingMovies        Contract  \
0                      No                   No  Month-to-month   
8                     Yes                  Yes  Month-to-month   
12                    Yes                  Yes        One year   
21    No internet service  No internet service        One year   
23                     No                   No        Two year   
...                   ...                  ...             ...   
7023                  Yes                  Yes  Month-to-month   
7024                   No                   No  Month-to-month   
7028                  Yes                   No        Two year   
7031                   No                   No        One year   
7041                   No                   No  Month-to-month   

     PaperlessBilling              PaymentMethod  MonthlyCharges  \
0                 Yes           Electronic check           29.85   
8                 Yes           Electronic check          104.80   
12                 No    Credit card (automatic)          100.35   
21                 No  Bank transfer (automatic)           19.80   
23                Yes    Credit card (automatic)           59.90   
...               ...                        ...             ...   
7023              Yes           Electronic check          103.50   
7024              Yes    Credit card (automatic)           84.

In [164]:
kel_d['TotalCharges'].sum()

5508865.75

#### kel_a = 5,283,104 (3280)
#### kel_b = 484,567 (359)
#### kel_c = 4,779,631.95 (1740)
#### kel_d = 5,508,865.75 (1653)

* --> jumlah total charges paling dipengaruhi oleh dependents, ini terbukti pada data jumlah total charges kelompok B yang mana mereka tidak memiliki partner akan tetapi memiliki tanggungan, sehingga income mereka dialokasikan untuk tanggungannya. Sedangkan untuk kelompok C, walaupun memiliki tanggungan, akan tetapi mereka memiliki partner, sehingga untuk charges dapat ditanggung bersama. 
* --> jumlah total charges terbesar, terdapat pada kelompok D dan A, yang mana kelommpok A tidak memiliki pasangan dan juga tanggungan, sedangkan untuk kelompok D yang memiliki jumlah total charges terbesar, mereka tidak memiliki tanggungan akan tetapi memiliki partner. Kelompok D dapat menjadi fakta bahwa memiliki partner tidak mengurangi jumlah total charges. 
* --> dapat dilihat juga dari jumlah customer pada masing-masing kelompok, untuk kelompok A terbanyak dikarenakan tidak memiliki partner dan tanggungan, untuk kelompok C dan D selisih <100, sedangkan untuk kelompok B dengan jumlah cutomer paling rendah. 